src/lib/RepositoryForms/Form/Processor/Content/UrlRedirectProcessor.php

```diff
@@ -117,26 +115,22 @@ private function resolveAdminSiteaccessRedirect(FormActionEvent $event): void
             return;
         }
 
-        /* If target URL was set to something else than Location, do nothing */
-        try {
-            $targetUrlAlias = $this->urlAliasService->lookup(
-                $response->getTargetUrl()
+        $location = $event->getOption('referrerLocation');
+
+        $targetUrl = $location instanceof Location
+            ? $this->router->generate(
+                '_ezpublishLocation',
+                [
+                    'locationId' => $location->getContentInfo()->mainLocationId,
```

Comment:
@alongosz You should redirect to `$location->id` not the main location.

Reply:
Good point... :sweat_smile: